### DISCUSSION TITLE
Include GitExtensionsShellEx*.dll files into portable version (necessary for Windows Explorer integration)

### DISCUSSION THREAD

**Discussion by gassanosh** [gitextensions/gitextensions#9690] [Q&A] [answered]
## Feature description

I usually use portable version of GitExtensions. But if I want to use Windows Explorer integration I have to download and install *.msi (and also configure it again as settings are stored in the different place).

To avoid those issues I have to use following workaround:
1. Download both *.msi and *.zip files
2. Use tool to extract GitExtensionsShellEx*.dll from *.msi
3. Add those files to portable version (*.zip file)

As for me it would be more convenient if the files were shipped with portable version. They are around 1 MB and the size mustn't be a problem.

## Environment

- GitExtensions version: 3.5.4.12724

**Comment by RussKie** [comment]
Windows Explorer integration requires shell extensions to be registered with Windows. This requires elevated privileges to register, and the assemblies must be at a known fixed location. A portable version isn't expected at be a fixed known location, because it is meant to be _portable_.

**Reply by gassanosh** [reply]
It's only one of possible usages of portable application. I agree, if you run app from usb stick there is no reason to register shell extension.
But if you have to work from different PCs and use many tools it's really a pain to keep all apps updated and consistent. 
But if you use portable apps you can keep all you data and binaries in the same place and easily sync them between all your PC and also setup new environment in minutes.
So I just sync all my portable apps between all PC and I don't have to update every application (and sync data) on every PC.
Anyway, what issues will you face if you add GitExtensionsShellEx*.dll to portable version? Did I miss something important?

**Comment by RussKie** [comment]
I hear what you're saying, but without an easy integration most users won't
know what to do with these files. This in turn may upset some users.
Your suggestion coupled with
https://github.com/gitextensions/gitextensions/pull/8010 would be a win
though.

**Reply by gassanosh** [reply]
https://github.com/gitextensions/gitextensions/issues/7982 looks interesting.
As https://github.com/gitextensions/gitextensions/pull/8010 is abandoned I opened another one https://github.com/gitextensions/gitextensions/pull/9701
Hope with this one you will agree to ship GitExtensionsShellEx*.dll with portable version

**Reply by gassanosh** [reply]
@RussKie Since #10421 is merged, can the issue be implemented? What should I do? Create a new issue or pull request?

**Reply by mstv** [reply, marked as answer]
Thank you for following up, Kirill.
The portable packages of v4.0.x do contain the `GitExtensionsShellEx32/64.dll`s.
